src/Components/ServiceScene/LogsVolumePanel.tsx

```diff
@@ -23,21 +41,59 @@ export class LogsVolumePanel extends SceneObjectBase<LogsVolumePanelState> {
         panel: this.getVizPanel(),
       });
     }
+
+    const fields = sceneGraph.lookupVariable(VAR_FIELDS, this) as AdHocFiltersVariable;
+    const patterns = sceneGraph.lookupVariable(VAR_PATTERNS, this) as CustomVariable;
+    const lineFilter = sceneGraph.lookupVariable(VAR_LINE_FILTER, this) as CustomVariable;
+    const labels = sceneGraph.lookupVariable(VAR_LABELS, this) as AdHocFiltersVariable;
+
+    this._subs.add(
+      fields.subscribeToState((newState, prevState) => {
+        if (newState.filters.length !== prevState.filters.length) {
```

Comment:
Filters may have the same length but different contents. What it's usually done, which is "not great", is comparing serialized strings with `JSON.parse()`.

Reply:
Helper function in https://github.com/grafana/explore-logs/pull/633 is being proposed as solution for this